[FrankenPHP] Add minimum memory threshold for worker restarts

karstennilsen opened the pull request:
## Summary
- Add `frankenphp_min_memory_mb` option to define minimum available memory (in MB) required before handling a new request
- If available memory falls below threshold after the first cycle, worker restarts (like `frankenphp_loop_max`)
- Configurable via `FRANKENPHP_MIN_MEMORY_MB` environment variable
- Default `-1` (disabled)
- Updated README with documentation

## How to test
- Set `FRANKENPHP_MIN_MEMORY_MB=64` and verify worker restarts when available memory drops below 64MB
- Verify worker continues normally when memory is sufficient
- Verify first request always processes regardless of memory

Commit:
Add a memory threshold setting for worker restarts

Introduce `frankenphp_min_memory_mb` option to enforce a minimum memory threshold before handling new requests. This prevents running workers when memory is critically low, enhancing stability. Option is configurable via environment variables or runtime options. Default value is -1 which disables this new functionality.

karstennilsen commented:
@Nyholm For our project this would be really helpful. What do you think about this?

Nyholm commented on src/frankenphp-symfony/src/Runner.php:
```diff
@@ -56,8 +57,40 @@ public function run(): int
+
+    private function hasAvailableMemory(int $loops): bool
+    {
+        if (-1 === $this->minMemoryMb || 1 === $loops) { // Always allow the first cycle
```
If `1 === $loops`, then we have completed the first request. If we are out of memory after the first request, we should restart, right? 

I suggest removing this check.

Nyholm commented on src/frankenphp-symfony/src/Runner.php:
```diff
@@ -56,8 +57,40 @@ public function run(): int
+        if (strlen($memoryLimit) < 2) {
+            return true; // Unexpected value, consider enough mem available
+        }
+        $last = \strtolower($memoryLimit[\strlen($memoryLimit) - 1]);
```
Note that a valid ini settings is: 

```
memory_limit=4711
```

This menas 4711 bytes. See https://www.php.net/manual/en/ini.core.php#ini.memory-limit

Nyholm reviewed (commented):
Thank you. Great idea. But I made some minor comments

Commit:
Implement feedback
- Skip first loop is not necessary since it's a do while (so always first cycle)
- Handle numeric-only memory_limit (bytes) in default case
